pkg/route/secretmanager/manager_test.go

```diff
@@ -0,0 +1,300 @@
+		},
+		{
+			name: "different routes can be registered with different secrets",
+			rs: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+				{routeName: "route2", secretName: "secret2"},
+			},
+			expectHandlersKeys: []string{namespace + "/route1", namespace + "/route2"},
+			expectErr:          0,
+		},
+		{
+			name: "error while adding SecretEventHandler",
+			rs: []routeSecret{
+				{routeName: "route", secretName: "secret"},
+			},
+			sm: fake.SecretMonitor{
+				Err: fmt.Errorf("some error"),
+			},
+			expectErr: 1,
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			mgr := NewManager(nil, nil).WithSecretMonitor(&s.sm)
+
+			gotErr := 0
+			for _, rs := range s.rs {
+				if err := mgr.RegisterRoute(context.TODO(), namespace, rs.routeName, rs.secretName); err != nil {
+					t.Log(err)
+					gotErr += 1
+				}
+			}
+			if gotErr != s.expectErr {
+				t.Fatalf("expected %d errors, got %d errors", s.expectErr, gotErr)
+			}
+			if len(s.expectHandlersKeys) != len(mgr.registeredHandlers) {
+				t.Fatalf("expected %d keys: %v, got %d keys: %v", len(s.expectHandlersKeys), s.expectHandlersKeys, len(mgr.registeredHandlers), mgr.registeredHandlers)
+			}
+			for _, key := range s.expectHandlersKeys {
+				if _, exists := mgr.registeredHandlers[key]; !exists {
+					t.Errorf("%s key should exist", key)
+				}
+			}
+		})
+	}
+}
+
+func TestUnregisterRoute(t *testing.T) {
+
+	type routeName string
+	namespace := "ns"
+
+	scenarios := []struct {
+		name               string
+		register           []routeSecret
+		unregister         []routeName
+		sm                 fake.SecretMonitor
+		expectHandlersKeys []string
+		expectErr          int
+	}{
+		{
+			name:       "unregister route without register",
+			register:   []routeSecret{},
+			unregister: []routeName{"route1"},
+			expectErr:  1,
+		},
+		{
+			name: "unregister route more than once",
+			register: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+			},
+			unregister: []routeName{"route1", "route1"},
+			expectErr:  1,
+		},
+		{
+			name: "error while removing SecretEventHandler",
+			register: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+			},
+			unregister: []routeName{"route1"},
+			sm: fake.SecretMonitor{
+				Err: fmt.Errorf("some error"),
+			},
+			expectHandlersKeys: []string{namespace + "/route1"},
+			expectErr:          1,
+		},
+		{
+			name: "correctly unregister route",
+			register: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+			},
+			unregister: []routeName{"route1"},
+			expectErr:  0,
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			mgr := NewManager(nil, nil)
+			// register
+			mgr.WithSecretMonitor(&fake.SecretMonitor{}) // avoid error from AddSecretEventHandler
+			for _, rs := range s.register {
+				if err := mgr.RegisterRoute(context.TODO(), namespace, rs.routeName, rs.secretName); err != nil {
+					t.Error(err)
```

Comment:
I don't think it makes sense to continue to the unregister function if register failed.
```suggestion
					t.Fatalf("failed to register %v: %v", rs, err)
```

Reply:
Correct, updated it